iterative/resource_runner.go

```diff
@@ -247,27 +247,30 @@ func resourceRunnerCreate(ctx context.Context, d *schema.ResourceData, m interfa
 
 	var logError error
 	var logEvents string
-	err = resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
-		switch cloud := d.Get("cloud").(string); cloud {
+	cloud := d.Get("cloud").(string)
+	ip := d.Get("instance_ip").(string)
+	err = resource.Retry(d.Timeout(schema.TimeoutCreate)-time.Minute, func() *resource.RetryError {
+
+		switch cloud {
 		case "kubernetes":
 			logEvents, logError = resourceMachineLogs(ctx, d, m)
 		default:
 			logEvents, logError = utils.RunCommand("journalctl --unit cml --no-pager",
 				2*time.Second,
-				net.JoinHostPort(d.Get("instance_ip").(string), "22"),
+				net.JoinHostPort(ip, "22"),
 				"ubuntu",
 				d.Get("ssh_private").(string))
 		}
 
-		log.Printf("[DEBUG] Collected log events: %#v", logEvents)
-		log.Printf("[DEBUG] Connection errors: %#v", logError)
-
 		if logError != nil {
-			return resource.RetryableError(fmt.Errorf("Waiting for the machine to accept connections... %s", logError))
-		} else if utils.HasStatus(logEvents, "terminated") {
-			return resource.NonRetryableError(fmt.Errorf("Failed to launch the runner!"))
-		} else if utils.HasStatus(logEvents, "ready") {
-			return nil
+			log.Printf("[DEBUG] Connection errors: %#v", logError)
```

Comment:
I take it with this we can see the reboot close the connection before the `cml.service` is started?

Reply:
Not sure what do you mean @dacbd. Could you please elaborate 🙏 ?

Reply:
@dacbd